src/Compilers/CSharp/Portable/BoundTree/VariablePendingInference.cs

```diff
@@ -58,12 +58,12 @@ internal BoundExpression SetInferredType(TypeSymbol type, Binder binderOpt, Diag
                                 ((DeclarationExpressionSyntax)this.Syntax).Type :
                                 this.Syntax;
 
-                            Binder.CheckRestrictedTypeInAsync(localSymbol.ContainingSymbol, type, diagnosticsOpt, typeOrDesignationSyntax);
+                            Binder.CheckRestrictedTypeInAsync(localSymbol.ContainingSymbol, type.TypeSymbol, diagnosticsOpt, typeOrDesignationSyntax);
                         }
                     }
 
-                    localSymbol.SetTypeSymbol(TypeSymbolWithAnnotations.Create(type));
```

Comment:
:memo: Previously, an `out var` would always be inferred to have non-nullable type.
It's not really clear how much nullability really matters in initial binding though. We'll have to follow-up on broader question.